SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt

```diff
@@ -47,22 +47,16 @@ class ApplicationPresenter: ApplicationContract.Presenter() {
         }
 
         launch {
-            lateinit var jsonString: DepartureDetails
-            try{
-                jsonString = client.get(apiCall)
-            } catch (e: Exception){
+            try {
+                var departureDetails: DepartureDetails = client.get(apiCall)
+                val departures: MutableList<DepartureInformation> = mutableListOf()
+                for (journey in departureDetails.outboundJourneys){
```

Comment:
This looks much nicer! Tiny nitpick though, you should have a space before your opening bracket here :)